i3ipc.py

```diff
@@ -413,6 +413,15 @@ def __init__(self, data, parent, conn):
             else:
                 setattr(self, attr, None)
 
+        # Make sure marks is never None (use []) to simplify client code. Older
+        # versions of i3 used mark instead of marks, so collapse that into
+        # marks to provide a consistent API.
+        if not self.marks:
```

Comment:
My thinking was that this would let people use the new i3ipc-python with an old version of i3, so they wouldn't have to upgrade the two in lockstep. With old versions of i3 the marks attribute will contain the 0 or 1 marks from i3's mark attribute.

If you prefer, I'm happy to remove mark entirely -- it would certainly simplify the code somewhat, and it probably isn't common that people want to update i3ipc-python ahead of i3.